PrivateClinicsWebNet.Application/Services/AppointmentService.cs

```diff
@@ -0,0 +1,127 @@
+using FluentValidation;
+using PrivateClinicsWebNet.Application.Abstractions;
+using PrivateClinicsWebNet.Application.DTOs.AppointmentDTOs;
+using PrivateClinicsWebNet.Application.Exceptions;
+using PrivateClinicsWebNet.BusinessLogic.Entities;
+using PrivateClinicsWebNet.Infrastructure.AppointmentInfrastructure.Abstractions;
+using PrivateClinicsWebNet.Infrastructure.AppointmentInfrastructure.DTOs;
+using PrivateClinicsWebNet.Infrastructure.Shared.Wrapper;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PrivateClinicsWebNet.Application.Services
+{
+    public class AppointmentService(IAppointmentRepository appointmentRepository,
+        IMapper mapper,
+        IValidator<AppointmentsRequestDto> requestValidator,
+        IValidator<CreateAppointmentDto> createDtoValidator,
+        IValidator<UpdateAppointmentDto> updateDtoValidator) : IAppointmentService
+    {
+        public async Task<Result<AppointmentResponseDto>> GetAppointmentAsync(string id)
+        {
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new InvalidAppointmentsIdException();
+            }
+
+            var response = await appointmentRepository.GetAppointmentByIdAsync(id);
+
+            if (response == null)
+            {
+                return Result<AppointmentResponseDto>.Failure($"The appointment with ID {id} wasn`t found.");
+            }
+
+            return await Result<AppointmentResponseDto>.SuccessAsync(response);
+        }
+
+        public async Task<Result<string>> CreateAppointmentAsync(CreateAppointmentDto appointmentDto)
+        {
+            var validationResult = createDtoValidator.Validate(appointmentDto);
+            
+            if (!validationResult.IsValid)
+            {
+                throw new InvalidAppointmentDataException();
+            }
+
+            var appointment = mapper.Map<Appointment>(appointmentDto);
+            var isCreatedResult = await appointmentRepository.СreateAppointmentAsync(appointment);
+
+            if (isCreatedResult)
+            {
+                return await Result<string>.SuccessAsync($"Appointment was created successfully: ID {appointment.Id}");
+            }
+
+            return Result<string>.Failure("Failed to create the appointment due to a system issue");
+        }
+
+        public async Task<Result<string>> UpdateAppointmentAsync(UpdateAppointmentDto appointmentDto)
+        {
+            var validationResult = updateDtoValidator.Validate(appointmentDto);
+            
+            if (!validationResult.IsValid)
+            {
+                throw new InvalidAppointmentDataException();
+            }
+
+            var appointment = mapper.Map<Appointment>(appointmentDto);
+            bool isExist = await appointmentRepository.IsAppointmentExistAsync(appointment.Id.ToString());
```

Comment:
AppointmetExistsAsync is better naming. and variable -> exists

Reply:
U can do the same approach with executeUpdate method to make ur queries more optimal